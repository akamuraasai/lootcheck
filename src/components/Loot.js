import React, { Component } from 'react';
import { connect } from 'react-redux';
import { acessaAPIBitcoin } from '../actions/bitcoin';

export class Loot extends Component {
  componentDidMount() {
    this.props.acessaAPIBitcoin();
  }

  computeBitcoin = () =>{
    const { bitcoin } = this.props;

    if (Object.keys(bitcoin).length === 0)
      return '';

    return this.props.saldo / bitcoin.bpi.BRL.rate_float;
  }

  render() {
    return (
      <h3>Saldo em Bitcoin: {this.computeBitcoin()}</h3>
    );
  }
}

export default connect(
  state => ({ saldo: state.saldo, bitcoin: state.bitcoin }),
  { acessaAPIBitcoin }
)(Loot);
