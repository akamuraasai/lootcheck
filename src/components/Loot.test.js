import React from 'react';
import { mount, shallow } from 'enzyme';
import { Loot } from './Loot';

describe('Loot', () => {
  let props = { saldo: 10, bitcoin: {} }
  let loot = shallow(<Loot {...props} />);

  it('renderiza corretamente', () => {
    expect(loot).toMatchSnapshot();
  });

  describe('quando montado', () => {
    const mockAcessaAPIBitcoin = jest.fn();
    beforeEach(() => {
      props.acessaAPIBitcoin = mockAcessaAPIBitcoin;
      loot = mount(<Loot {...props} />);
    });

    it('dispara a função "acessaAPIBitcoin" recebida via props', () => {
      expect(mockAcessaAPIBitcoin).toHaveBeenCalled()
    });
  });

  describe('quando existem valores válidos na props "bitcoin"', () => {
    beforeEach(() => {
      props = { saldo: 10, bitcoin: { bpi: { BRL: { rate_float: 999.99 } } } };
      loot = shallow(<Loot {...props} />);
    });

    it('mostra o valor correto do bitcoin', () => {
      const calc = props.saldo / props.bitcoin.bpi.BRL.rate_float;
      expect(loot.find('h3').text()).toEqual('Saldo em Bitcoin: ' + calc);
    });
  });
});
