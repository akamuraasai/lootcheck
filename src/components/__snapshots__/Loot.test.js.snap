// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot renderiza corretamente 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Loot
    bitcoin={Object {}}
    saldo={10}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Saldo em Bitcoin: ",
        "",
      ],
    },
    "ref": null,
    "rendered": Array [
      "Saldo em Bitcoin: ",
      "",
    ],
    "type": "h3",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Saldo em Bitcoin: ",
          "",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Saldo em Bitcoin: ",
        "",
      ],
      "type": "h3",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
